classes/services/NavigationMenuService.inc.php

```diff
@@ -44,11 +52,45 @@ public function getMenuItemTypesCallback($hookName, $args) {
 				'title' => __('navigation.catalog'),
 				'description' => __('navigation.navigationMenus.catalog.description'),
 			),
+			NMI_TYPE_SERIES => array(
+				'title' => __('navigation.navigationMenus.series.generic'),
+				'description' => __('navigation.navigationMenus.series.description'),
+			),
+			NMI_TYPE_CATEGORY => array(
+				'title' => __('navigation.navigationMenus.category.generic'),
+				'description' => __('navigation.navigationMenus.category.description'),
+			),
+			NMI_TYPE_NEW_RELEASE => array(
+				'title' => __('navigation.navigationMenus.newRelease'),
+				'description' => __('navigation.navigationMenus.newRelease.description'),
+			),
```

Comment:
So, thinking about the messages you display when there are no series or categories, (eg - `manager.navigationMenus.form.navigationMenuItem.series.noItems`). How difficult would it be to perform that check here? You can then just not add the series or category option to the selection when there are no series or categories. That will keep the list smaller for users that don't have series or categories, and it may simplify some of your template and validation logic since you won't have to account for people selecting an option they can't use.